src/SLIP10Node.ts

```diff
@@ -0,0 +1,278 @@
+   */
+  public async derive(path: SLIP10PathTuple): Promise<SLIP10Node> {
+    const { key, depth } = await deriveChildNode(
+      this.keyBuffer,
+      this.depth,
+      path,
+      this.curve,
+    );
+
+    return new SLIP10Node({
+      key,
+      depth,
+      curve: this.curve,
+    });
+  }
+
+  // This is documented in the interface of this class.
+  toJSON(): JsonSLIP10Node {
+    return {
+      depth: this.depth,
+      key: this.key,
+    };
+  }
+}
+
+async function createKeyFromPath({
+  derivationPath,
+  depth,
+  key,
+  curve,
+}: Required<Pick<SLIP10NodeOptions, 'derivationPath'>> &
+  Omit<SLIP10NodeOptions, 'derivationPath'>) {
+  if (key) {
+    throw new Error(
+      'Invalid parameters: May not specify a derivation path if a key is specified. Initialize the node with just the parent key and its depth, then call BIP44Node.derive() with your desired path.',
+    );
+  }
+
+  if (depth) {
+    throw new Error(
+      'Invalid parameters: May not specify a depth if a derivation path is specified. The depth will be calculated from the path.',
+    );
+  }
+
+  if (derivationPath.length === 0) {
+    throw new Error(
+      'Invalid derivation path: May not specify an empty derivation path.',
+    );
+  }
+
+  const _depth = derivationPath.length - 1;
+  validateBIP32Depth(_depth);
+
+  return await deriveKeyFromPath(derivationPath, undefined, _depth, curve);
+}
+
+/**
+ * Validates the curve.
+ * @param curve
+ */
+function validateCurve(curve: Curve): asserts curve is Curve {
+  if (!curve) {
+    throw new Error('Invalid curve: Must specify a curve.');
+  }
+
+  if (!SUPPORTED_CURVES.includes(curve.name)) {
+    throw new Error(
+      `Invalid curve: Only the following curves are supported: ${SUPPORTED_CURVES.join(
+        ', ',
+      )}.`,
+    );
+  }
+}
+
+/**
+ * Validates a BIP-32 path depth. Effectively, asserts that the depth is an
+ * integer `number`. Throws an error if validation fails.
+ *
+ * @param depth - The depth to validate.
+ */
+export function validateBIP32Depth(depth: unknown): asserts depth is number {
+  if (typeof depth !== 'number' || !Number.isInteger(depth) || depth < 0) {
+    throw new Error(
+      `Invalid HD tree path depth: The depth must be a positive integer. Received: "${depth}"`,
+    );
+  }
+}
+
+/**
+ * Derives a child key from the given parent key.
+ * @param parentKey - The parent key to derive from.
+ * @param parentDepth - The depth of the parent key.
+ * @param pathToChild - The path to the child node / key.
+ * @param curve - The curve to use.
+ * @returns The derived key and depth.
+ */
+export async function deriveChildNode(
+  parentKey: Buffer,
+  parentDepth: number,
+  pathToChild: SLIP10PathTuple,
+  curve: Curve,
+): Promise<{ key: Buffer; depth: number }> {
+  if (pathToChild.length === 0) {
+    throw new Error(
+      'Invalid HD tree derivation path: Deriving a path of length 0 is not defined',
```

Comment:
Consistency is king:
```suggestion
      'Invalid HD tree derivation path: Deriving a path of length 0 is not defined.',
```